massa-network-worker/src/peer_info_database.rs

```diff
@@ -336,6 +336,78 @@ impl PeerInfoDatabase {
         Ok(())
     }
 
+    pub async fn whitelist(&mut self, ips: Vec<IpAddr>) -> Result<(), NetworkError> {
+        for ip in ips.into_iter() {
+            let old_pt = if let Some(peer) = self.peers.get_mut(&ip) {
+                let pt = peer.peer_type;
+                if pt != PeerType::WhiteListed {
+                    peer.peer_type = PeerType::WhiteListed;
+                }
+                pt
+            } else {
+                let mut p = PeerInfo::new(ip, false);
+                p.peer_type = PeerType::WhiteListed;
+                self.peers.insert(ip, p);
+                PeerType::WhiteListed
+            };
+
+            if old_pt == PeerType::WhiteListed {
+                continue;
+            }
+            // update global connection counts by peer type
+            let peer = *self.peers.get(&ip).unwrap(); // filled just before
+            if peer.active_out_connection_attempts > 0 {
+                self.decrease_global_active_out_connection_attempt_count(old_pt, &ip)?;
+                self.increase_global_active_out_connection_attempt_count(
```

Comment:
When you create a new peer L.349 you want to call this function ? Because here it's not trigger and the next ones that increase the whitelist count also.

Reply:
I don't understand your comment

Reply:
When you create a peer in this function you set the old peer type to Whitelisted so we will not go into this condition but if you have a new Whitelist peer you want to increase the count of whitelist peers no ?

Reply:
old peer type is never set to whitelist. If the peer was whitelist before, there is no need to update the global counters but if it was something else we want to decrease the old peer type counters and increase the whitelist ones

Reply:
L.351 after create a new peer if it don't exist the old peer type is set to whitelist.